src/amqpvalue.c

```diff
@@ -2674,6 +2674,60 @@ static int encode_long(AMQPVALUE_ENCODER_OUTPUT encoder_output, void* context, i
 	return result;
 }
 
+static int encode_float(AMQPVALUE_ENCODER_OUTPUT encoder_output, void* context, float value)
+{
+    int result;
+
+    uint32_t value_as_uint32 = *((uint32_t*)(void*)&value);
```

Comment:
This encoding is not following the IEEE 754 algorithm.
What about the fraction part?